drivers/usb/device/usb_dc_mcux.c

```diff
@@ -275,7 +275,7 @@ int usb_dc_ep_configure(const struct usb_dc_ep_cfg_data *const cfg)
 		block->data = NULL;
 	}
 
-	block->data = k_heap_alloc(&ep_buf_pool, cfg->ep_mps, K_NO_WAIT);
+	block->data = k_heap_alloc(&ep_buf_pool, CONFIG_USB_MCUX_ENDPOINT_BUFFER_SIZE, K_NO_WAIT);
```

Comment:
Pool is defined by `K_HEAP_DEFINE(ep_buf_pool, 1024 * EP_BUF_NUMOF_BLOCKS);`
Can we remove K_HEAP_DEFINE and k_heap_alloc(), change `struct k_mem_block block;` to `uint8_t *block;` and define/assign static buffers to `s_ep_ctrl[]` using `LISTIFY(NUM_OF_EP_MAX, ....)`?

Reply:
I did have the same thought but I wasn't sure if it was appropriate for this PR, I suppose I can knock out two birds with one stone though?